packages/calypso-e2e/src/lib/blocks/block-flows/subscribe.ts

```diff
@@ -21,7 +21,7 @@ export class SubscribeFlow implements BlockFlow {
 			.getByRole( 'textbox', { name: 'Type your email' } );
 
 		// The email input field only appears if not logged in, or if logged in user is not subscribed already.
-		if ( emailInputLocator ) {
+		if ( emailInputLocator && ( await emailInputLocator.isVisible() ) ) {
```

Comment:
I think you'll never get something that's not a Locator object, so this could be simplified to just
```suggestion
		if ( await emailInputLocator.isVisible() ) {
```